pkg/controller/psrestore/controller.go

```diff
@@ -162,6 +165,30 @@ func (r *PerconaServerMySQLRestoreReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, nil
 	}
 
+	restoreList := &apiv1alpha1.PerconaServerMySQLRestoreList{}
+	if err := r.List(ctx, restoreList, &client.ListOptions{Namespace: cr.Namespace}); err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "get restore jobs list")
+	}
+	for _, restore := range restoreList.Items {
+		if restore.Spec.ClusterName == cr.Spec.ClusterName && restore.Name != cr.Name {
```

Comment:
I think it'd be more readable if you do negative check and continue if cluster name is different or restore name is the same

Reply:
we can lose an indentation level this way

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/382/commits/fde424dedeec1830a8cbd0c03b17b0f6764f64c4